Automattic/wp-calypso
# Add-Ons: Remove billing-transactions state and use purchases state instead

Fixes https://github.com/Automattic/wp-calypso/issues/87529
Fixes https://github.com/Automattic/wp-calypso/issues/86147

## Proposed Changes

Part of uncoupling the add-on metadata from Calypso dependencies.

- Removes the billing-transactions state for determining if an add-on was purchased. Uses the purchases state instead (and the new data-stores hook we've put together in a parent PR).
    - I cannot see what the need was to use billing-transactions previously. We had previous logic for determining the purchase status for add-ons that are used in the UI. Perhaps that needed to be expanded instead. I cannot see why it needs to be fundamentally different for storage (meaning, to derive it from the billing-log). I could be missing something though 🤔 
- Cleans up the code a bit and adds several TODO comments for a follow-up. The logic for determining the "purchase status" of add-ons is a bit scattered now, with storage/space add-ons doing it differently and separately. We ought to at least explore further unifying this part of the logic.
- Removes the `use-billing-transactions` hook as an orphan/unused code now. 
    - it was also a bit suspiciously living under `state/sites` when nothing really site-specific about it (it's querying `/me/billing-history/past`)? 
    - ~on closer look, I wonder if this was causing other problems with multiple user sites - meaning, whether purchasing a space add-on for one site would show it as "purchased" for another site~ (I can see they were handling this via further "filters" in the add-ons code that consider the site ID)

## Media

### After removing storage add-on purchase

**Before (cannot repurchase)**

<img width="500" alt="Screenshot 2024-03-11 at 2 52 34 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/05c4a30f-606a-49a0-86c5-8f69a3471558">


**After**

<img width="500" alt="Screenshot 2024-03-11 at 2 52 39 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/3d80f9a5-c14f-4fb1-95ae-96f5c8edc162">


## Testing Instructions



* Go to `/add-ons/[ site ]` and confirm storage add-ons are visible and purchasable
* Confirm that we can purchase, cancel & remove subscription, and re-purchase the same storage add-on (e.g. 50GB)
* Confirm any other criteria for showing storage add-ons works as previously e.g. 100GB not shown if it would exceed the site's limits (?)
* Confirm add-ons in Signup work as expected (compare with production). The signup flow is `/start/with-add-ons` (select a paid domain, then select a plan, then the next page should be the add-ons)

## Pre-merge Checklist
